src/audio/dai-zephyr.c

```diff
@@ -1227,9 +1227,23 @@ static int dai_copy(struct comp_dev *dev)
 	/* get data sizes from DMA */
 	ret = dma_get_status(dd->chan->dma->z_dev, dd->chan->index, &stat);
 	if (ret < 0) {
-		dai_report_xrun(dev, 0);
+		/* DMA status can return -EPIPE and current status content if xrun occurs */
+		if (ret == -EPIPE)
+#if CONFIG_NOTIFY_HOST_ON_XRUN
+			dai_report_xrun(dev, 0);
 		return ret;
+#else
+			if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
+				comp_warn(dev, "dai_copy(): dma_get_status() underrun occurred, "
+					  "ret = %u", ret);
+			else
+				comp_warn(dev, "dai_copy(): dma_get_status() overrun occurred, "
+					  "ret = %u", ret);
```

Comment:
I think you need braces here. GCC used to complain about constructs like
```
if (a)
	if (b)
		do_x();
	else
		do_y();
```
because of the potential ambiguity of the `else` (this ain't no python). This probably wasn't compiled with `CONFIG_NOTIFY_HOST_ON_XRUN` disabled